networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRangeMessageHandler.java

```diff
@@ -0,0 +1,299 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.networking.eth2.rpc.beaconchain.methods;
+
+import static tech.pegasys.teku.spec.config.Constants.MIN_EPOCHS_FOR_BLOB_SIDECARS_REQUESTS;
+
+import com.google.common.base.Throwables;
+import java.nio.channels.ClosedChannelException;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes32;
+import org.hyperledger.besu.plugin.services.MetricsSystem;
+import org.hyperledger.besu.plugin.services.metrics.Counter;
+import org.hyperledger.besu.plugin.services.metrics.LabelledMetric;
+import org.jetbrains.annotations.NotNull;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.metrics.TekuMetricCategory;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.networking.eth2.peers.Eth2Peer;
+import tech.pegasys.teku.networking.eth2.rpc.core.PeerRequiredLocalMessageHandler;
+import tech.pegasys.teku.networking.eth2.rpc.core.ResponseCallback;
+import tech.pegasys.teku.networking.eth2.rpc.core.RpcException;
+import tech.pegasys.teku.networking.p2p.rpc.StreamClosedException;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.datastructures.blocks.SignedBeaconBlock;
+import tech.pegasys.teku.spec.datastructures.execution.versions.deneb.BlobSidecar;
+import tech.pegasys.teku.spec.datastructures.networking.libp2p.rpc.BlobSidecarsByRangeRequestMessage;
+import tech.pegasys.teku.storage.client.CombinedChainDataClient;
+import tech.pegasys.teku.storage.client.RecentChainData;
+
+public class BlobSidecarsByRangeMessageHandler
+    extends PeerRequiredLocalMessageHandler<BlobSidecarsByRangeRequestMessage, BlobSidecar> {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final Spec spec;
+  private final UInt64 denebForkEpoch;
+  private final RecentChainData recentChainData;
+  private final CombinedChainDataClient combinedChainDataClient;
+  private final UInt64 maxRequestSize;
+  private final UInt64 maxBlobsPerBlock;
+  private final LabelledMetric<Counter> requestCounter;
+  private final Counter totalBlobSidecarsRequestedCounter;
+
+  public BlobSidecarsByRangeMessageHandler(
+      final Spec spec,
+      final UInt64 denebForkEpoch,
+      final MetricsSystem metricsSystem,
+      final RecentChainData recentChainData,
+      final CombinedChainDataClient combinedChainDataClient,
+      final UInt64 maxRequestSize,
+      final UInt64 maxBlobsPerBlock) {
```

Comment:
Dmitry is right. You're statically initializing this handler with deneb config, while this parameter is design to be adjusted in new milestones, so it will change at runtime when transitioning to future milestones. Not required now.

Reply:
I'll consider this OK for now then